library/psa_crypto_rsa.c

```diff
@@ -172,6 +172,9 @@ psa_status_t mbedtls_psa_rsa_export_key(psa_key_type_t type,
                                         size_t data_size,
                                         size_t *data_length)
 {
+/* Currently this is always true due to the fact that build_info.h auto-enables
+ * MBEDTLS_PK_WRITE_C whenever both PSA and RSA_C are enabled.
+ * However we keep the guard here to emphasize this dependency explicitly. */
```

Comment:
Sorry, I changed my mind since my last comment: the bits in `build_info.h` are not something that's likely to change. PSA capabilities are determined by `PSA_WANT` macros. As soon as we have RSA keys enabled, we must be able to export them, because there's no way for `PSA_WANT` to express "built-in RSA but we can't export keys".

Also, this PR removes the `PK_WRITE` dependency declaration from tests that export RSA keys, which is fully consistent with the above rule: PSA tests should only have PSA dependencies, not `MBEDTLS_` ones.

So, we need to be consistent and remove the `#if` here: whenever we have built-in RSA, we must have a working `mbedtls_psa_rsa_export_key()`.